web/forms/implementing_simple_autocompletion/match_element.dart

```diff
@@ -1,34 +1,23 @@
 @HtmlImport('match_element.html')
-library web.forms.implementing_simple_autocompletion_match;
-
-import 'dart:html';
+library match_element;
 
+import 'package:web_components/web_components.dart' show HtmlImport;
 import 'package:polymer/polymer.dart';
 
-@CustomTag('match-element')
+@PolymerRegister('match-element')
 class MatchElement extends PolymerElement {
-
   /// Candidate string.
-  ///
   /// The matching string which we want to display highlighted.
-  @published String value;
+  @Property(observer: 'valueChanged') String value;
 
   /// Input query.
-  ///
   /// The query whose first match in `value` we want to display highlighted.
-  @published String inputQuery;
-
-  NodeValidator nodeValidator;
+  @Property(observer: 'valueChanged') String inputQuery = '';
 
-  MatchElement.created() : super.created() {
-    nodeValidator = new NodeValidatorBuilder()
-      ..allowTextElements()
-      ..allowElement('strong');
-  }
+  MatchElement.created() : super.created();
 
-  void valueChanged(String old) {
-    injectBoundHtml(highlightedValue, element: $['container'], validator: nodeValidator);
-  }
+  @eventHandler
```

Comment:
Might be a good candidate for a multi-property observer: 

```dart
@Observe('value, inputQuery')
valueChanged(_, __) => ...;
```

Then you could remove both of the observers in the `@Property` annotations above.